### PR TITLE
Prometheus helm chart upgrade

### PR DESCRIPTION
**Pull request by hisarbalik**
Is upgrading the monitoring component to the latest version of the prometheus-operator.

- All customization is done only in values.yaml
- kyma-specific resources are located in kyma-* specific folders.
- prometheus-operator component removed
- all relevant CRDs moved to cluster-essentials
- alertrules are consolidated

Related issues:
#4520 #6124

This PR is not covering the consolidation of servicemonitors and dashboards. That will happen in follow-up PRs very soon in order to have a simpler review

### CODE REVIEW

File: resources/monitoring/charts/prometheus-operator/admission-webhooks/job-patch/psp.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    rule: 'RunAsAny'
+  seLinux:
+    # This policy assumes the nodes are using AppArmor rather than SELinux.
+    rule: 'RunAsAny'
+  supplementalGroups:
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 0
+        max: 65535
```
Comment (zralt) [inline resources/monitoring/charts/prometheus-operator/admission-webhooks/job-patch/psp.yaml]: This actually enables to run as root

File: resources/monitoring/charts/prometheus-operator/admission-webhooks/job-patch/psp.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 0
+        max: 65535
```
Comment (zralt) [inline resources/monitoring/charts/prometheus-operator/admission-webhooks/job-patch/psp.yaml]: This actually enables to run as root

File: resources/monitoring/charts/prometheus-operator/psp.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    rule: 'RunAsAny'
+  seLinux:
+    # This policy assumes the nodes are using AppArmor rather than SELinux.
+    rule: 'RunAsAny'
+  supplementalGroups:
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 0
+        max: 65535
```
Comment (zralt) [inline resources/monitoring/charts/prometheus-operator/psp.yaml]: This actually enables root

File: resources/monitoring/charts/prometheus-operator/psp.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+      # Forbid adding the root group.
+      - min: 0
+        max: 65535
```
Comment (zralt) [inline resources/monitoring/charts/prometheus-operator/psp.yaml]: This actually enables root

**Comment by derberg** [comment]
I checked on a cluster and all dashboards display data correctly as expected. Thanks

File: resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-per-client-dashboard.json [hunk 5]
```diff
@@ -1,446 +0,0 @@
-{
-  "annotations": {
-    "list": [
```
Comment (piotrmsc) [inline resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-per-client-dashboard.json]: I cannot find this chart in the new directory & on the provisioned cluster for review, is there a replacement for it?

File: resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-dashboard.json [hunk 6]
```diff
@@ -1,859 +0,0 @@
-{
```
Comment (piotrmsc) [inline resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-dashboard.json]: I see new apiserver dashboards but I cannot find this one with general http metrics, is there a replacement for it?

File: resources/monitoring/charts/grafana/dashboards/apiserver-proxy.json [hunk 7]
```diff
@@ -1,781 +0,0 @@
...
-      "id": 16,
-      "legend": {
-        "avg": false,
-        "current": false,
-        "max": false,
-        "min": false,
-        "show": true,
-        "total": false,
-        "values": false
-      },
```
Comment (piotrmsc) [inline resources/monitoring/charts/grafana/dashboards/apiserver-proxy.json]: It was deleted and it's not in the new directory. This chart is for apiserver proxy - not apiserver itself

File: resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-per-client-dashboard.json (hunk 5 above)
Comment (hisarbalik) [reply resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-per-client-dashboard.json]: is this a custom dashboard?

File: resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-dashboard.json (hunk 6 above)
Comment (hisarbalik) [reply resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-dashboard.json]: there is some changes in the original chart, if we still need old dashboard we can add as a custom chart to the kyma

**Comment by janmedrek** [comment]
`Application Registry` and `Connector Service` dashboards are missing their custom metrics, will try to check whether it's Prometheus or access - related.

File: resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-dashboard.json (hunk 6 above)
Comment (a-thaler) [reply resources/monitoring/charts/grafana/dashboards/kubernetes-apiserver-dashboard.json]: I added the old chart

**Comment by a-thaler** [comment]
@janmedrek I think there was simply traffic missing. I just caused some traffic on registry and connector service and the chart filled, please check again

**Comment by pkosiec** [comment]
Before merge, please also update the file `installation/resources/installer-cr-cluster-compass.yaml.tpl`, as in a recent PR #6240 the prometheus-operator has been introduced there. Thanks!

**Pull request merged**